### PR TITLE
feat(feature-activation): add feature settings validation

### PR DESCRIPTION
**Pull request by glevco**
### Motivation

This PR implements a validation that compares the Feature Activation settings from the configuration file, against the previous value that is stored in the database. This runs during full node initialization and is just to prevent human errors when editing those files, since one cannot remove or change existing features from there.

### Acceptance Criteria

- Create new `FeatureActivationStorage` that validates configured feature activation settings against the previous value from the database.
- Add new `reset-feature-settings` CLI tool for erasing saved feature settings.

### Checklist

- [x] If you are requesting a merge into `master`, confirm this code is production-ready and can be included in future releases as soon as it gets merged

### CODE REVIEW

File: hathor/feature_activation/feature.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 
 @unique
-class Feature(Enum):
+class Feature(str, Enum):
```
Comment (msbrogli) [inline hathor/feature_activation/feature.py:19]: Why this change?

File: hathor/manager.py [hunk 2]
```diff
@@ -98,6 +99,7 @@ def __init__(self,
                  event_manager: EventManager,
                  feature_service: FeatureService,
                  bit_signaling_service: BitSignalingService,
+                 feature_storage: FeatureActivationStorage | None,
```
Comment (msbrogli) [inline hathor/manager.py:102]: Why optional? Shouldn't it be part of the `BitSignalingService`?

File: hathor/feature_activation/storage/feature_activation_storage.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+
+class FeatureActivationStorage:
+    __slots__ = ('_log', '_settings', '_db', '_cf_meta')
+
+    def __init__(self, *, settings: HathorSettings, rocksdb_storage: RocksDBStorage) -> None:
+        self._log = logger.new()
+        self._settings = settings
+        self._db = rocksdb_storage.get_db()
+        self._cf_meta = rocksdb_storage.get_or_create_column_family(_CF_NAME_META)
```
Comment (msbrogli) [inline hathor/feature_activation/storage/feature_activation_storage.py:36]: Should we use a single column family for all settings related stuff?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: hathor/feature_activation/feature.py (hunk 1 above)
Comment (glevco) [reply hathor/feature_activation/feature.py:19]: Because now we're serializing this to json (as it's stored in the database), so adding this make it work (otherwise it tries to serialize a raw enum and fails). It's equivalent to using `StrEnum` from Python 3.11.

File: hathor/manager.py (hunk 2 above)
Comment (glevco) [reply hathor/manager.py:102]: It's optional because when the full node is running with `--memory-storage`, this check doesn't make sense. As this is only here to prevent human error in the configuration file, it's only necessary when running the full node "for real", with a rocksdb storage.

Also, I left it separate from the `BitSignalingService` for the same reason, as this is just a pre-initialization check. Maybe the name `FeatureActivationStorage` is not so good, it could be something more specific like `FeatureActivationSettingsValidator`?

File: hathor/feature_activation/storage/feature_activation_storage.py (hunk 3 above)
Comment (glevco) [reply hathor/feature_activation/storage/feature_activation_storage.py:36]: Makes sense, but what other related stuff do you mean? Like using it to store the whole `HathorSettings` instance?

File: hathor/feature_activation/feature.py (hunk 1 above)
Comment (msbrogli) [reply hathor/feature_activation/feature.py:19]: Got it. Thanks!

File: hathor/manager.py (hunk 2 above)
Comment (msbrogli) [reply hathor/manager.py:102]: Got it. But why would HathorManager have to understand the validation of the feature service? Shouldn't the FeatureService or the BitSignalingService do it themselves? Maybe they will have to have a `start()` method?

Comment (glevco) [reply hathor/manager.py]: I moved it to the `BitSignalingService` on https://github.com/HathorNetwork/hathor-core/pull/894/commits/dfb7c66a17ae0452453ea1ffe4bbb9307a19676e

**Review by jansegre — APPROVED** [review]

File: hathor/manager.py (hunk 2 above)
Comment (msbrogli) [reply hathor/manager.py]: Thanks!

File: hathor/feature_activation/storage/feature_activation_storage.py (hunk 3 above)
Comment (msbrogli) [reply hathor/feature_activation/storage/feature_activation_storage.py:36]: Now the whole HathorSettings but we do have some flags stored in the database here and there in the code.

**Review by msbrogli — DISMISSED** [review]

File: hathor/feature_activation/storage/feature_activation_storage.py (hunk 3 above)
Comment (msbrogli) [reply hathor/feature_activation/storage/feature_activation_storage.py:36]: I guess we don't have to do this refactor in this PR. So I'll approve this PR and we can discuss this refactor separately. Thanks!

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]

**Review by jansegre — DISMISSED** [review]

**Review by msbrogli — APPROVED** [review]

**Review by jansegre — APPROVED** [review]
